internal/controllers/opentelemetrycollector_controller.go

```diff
@@ -219,6 +219,8 @@ func NewReconciler(p Params) *OpenTelemetryCollectorReconciler {
 // +kubebuilder:rbac:groups=coordination.k8s.io,resources=leases,verbs=get;list;create;update
 // +kubebuilder:rbac:groups=monitoring.coreos.com,resources=servicemonitors;podmonitors,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=networking.k8s.io,resources=ingresses,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=networking.k8s.io,resources=networkpolicies,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=apps,resources=deployments/finalizers,verbs=get;watch;update;patch
```

Comment:
Why did we add `deployments/finalizers`?

Reply:
This is needed because the NP has ownerreference to the operator deployment for proper cleanup.

Otherwise I was getting 

```
error creating NetworkPolicy for the operator: networkpolicies.networking.k8s.io "opentelemetry-operator" is forbidden: cannot set blockOwnerDeletion if an ownerReference refers to a resource you can't set finalizers on: , <nil>
```